src/database/database.h

```diff
@@ -32,6 +32,7 @@ class Database
 	virtual void beginSave() = 0;
 	virtual void endSave() = 0;
 	virtual bool initialized() const { return true; }
+	virtual void verifyDatabase() = 0;
```

Comment:
I think `pingConnection` would be a better name.
Also an empty default implementation would be nice.

Reply:
the problem is that the ping is just the pg name, in other backend it's not the same name but i can rename if you prefer
i can try an empty default, it was on a previous first try, maybe if it works i can remove upper classes empty body but not sure du to compiler ambigous

Reply:
The issue I have with the name is that "verify the database" sounds a lot like you would be checking the consistency of correctness of the database structures or contents.

Reply:
okay then i move to ping